File: MyInflater/app/src/main/java/com/moonayoung/inflater/MainActivity.java
package com.moonayoung.inflater;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main); //R.layout.activity_main: res폴더 밑에 layout폴더 밑에 activity_main.xml 가리킴
        //메모리에 버튼이 만들어지는 자체가 이 함수 이후로 나타나는 거임. 밑으로 가면 버튼 못 찾음.
        Button button=findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {

            }
        });
    }
}

File: Kotlin_Socket/app/src/main/java/com/moonayoung/kotlin_socket/MainActivity.kt
package com.moonayoung.kotlin_socket

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*;
import java.io.ObjectInputStream
import java.io.ObjectOutputStream
import java.net.Socket

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        sendButton.setOnClickListener {
            var data : String = input1.text.toString()

            Thread(Runnable {
                kotlin.run {
                    send(data)
                }
            })
        }
    }

    fun send(data : String){
        var port = 5001

        try {
            var sock : Socket = Socket("localhost", port)

            var outStream : ObjectOutputStream(sock.getInputStream)
            outStream.writeObject(data)
            outStream.flush()

            var inputStream : ObjectInputStream(sock)
        }
    }

}